tests/rpmsigdig.at

```diff
@@ -1093,6 +1100,275 @@ echo $?
+],
+[0],
+[/tmp/hello-2.0-1.x86_64.rpm:
+    Header V4 RSA/SHA512 Signature, key ID 1964c5fc: OK
+    Header SHA256 digest: OK
+    Header SHA1 digest: OK
+    Payload SHA256 digest: OK
+    MD5 digest: OK
+],
+[])
+
+RPMTEST_CHECK([
+runroot rpmsign --addsign --rpmv6 --key-id 152BB32FD9CA982797E835CFB0645AEC757BF69E /tmp/hello-2.0-1.x86_64.rpm &> /dev/null
+runroot rpmkeys -Kv /tmp/hello-2.0-1.x86_64.rpm
+],
+[0],
+[/tmp/hello-2.0-1.x86_64.rpm:
+    Header V4 EdDSA/SHA512 Signature, key ID 757bf69e: OK
+    Header V4 RSA/SHA512 Signature, key ID 1964c5fc: OK
+    Header SHA256 digest: OK
+    Header SHA1 digest: OK
+    Payload SHA256 digest: OK
+    MD5 digest: OK
+],
+[])
+
+RPMTEST_CHECK([
+runroot rpmsign --addsign --rpmv6 --key-id E8A62C0512B06B5D2183BA207F1C21F95F65BBE8 /tmp/hello-2.0-1.x86_64.rpm &> /dev/null
+runroot rpmkeys -Kv /tmp/hello-2.0-1.x86_64.rpm
+],
+[0],
+[/tmp/hello-2.0-1.x86_64.rpm:
+    Header V4 ECDSA/SHA512 Signature, key ID 5f65bbe8: OK
+    Header V4 EdDSA/SHA512 Signature, key ID 757bf69e: OK
+    Header V4 RSA/SHA512 Signature, key ID 1964c5fc: OK
+    Header SHA256 digest: OK
+    Header SHA1 digest: OK
+    Payload SHA256 digest: OK
+    MD5 digest: OK
+],
+[])
+
+RPMTEST_CHECK([
+runroot rpmkeys --delete 757bf69e-661d22a8
+runroot rpmkeys -Kv /tmp/hello-2.0-1.x86_64.rpm
+],
+[1],
+[/tmp/hello-2.0-1.x86_64.rpm:
+    Header V4 ECDSA/SHA512 Signature, key ID 5f65bbe8: OK
+    Header V4 EdDSA/SHA512 Signature, key ID 757bf69e: NOKEY
+    Header V4 RSA/SHA512 Signature, key ID 1964c5fc: OK
+    Header SHA256 digest: OK
+    Header SHA1 digest: OK
+    Payload SHA256 digest: OK
+    MD5 digest: OK
+],
+[])
+
+RPMTEST_CHECK([
+runroot rpmsign --resign --key-id E8A62C0512B06B5D2183BA207F1C21F95F65BBE8 /tmp/hello-2.0-1.x86_64.rpm &> /dev/null
+runroot rpmkeys -Kv /tmp/hello-2.0-1.x86_64.rpm
+],
+[0],
+[/tmp/hello-2.0-1.x86_64.rpm:
+    Header V4 ECDSA/SHA512 Signature, key ID 5f65bbe8: OK
+    Header SHA256 digest: OK
+    Header SHA1 digest: OK
+    Payload SHA256 digest: OK
+    MD5 digest: OK
+],
+[])
+
+RPMTEST_CHECK([
+runroot rpmsign --delsign /tmp/hello-2.0-1.x86_64.rpm &> /dev/null
+runroot rpmkeys -Kv /tmp/hello-2.0-1.x86_64.rpm
+],
+[0],
+[/tmp/hello-2.0-1.x86_64.rpm:
+    Header SHA256 digest: OK
+    Header SHA1 digest: OK
+    Payload SHA256 digest: OK
+    MD5 digest: OK
+],
+[])
+
+RPMTEST_CLEANUP
+
+AT_SETUP([rpmsign --addsign multisig v6])
+AT_KEYWORDS([rpmsign signature multisig v6])
+AT_SKIP_IF([test x$PGP = xdummy])
+RPMDB_INIT
+
+RPMTEST_CHECK([
+RPM_DOCDIR=$(runroot rpm --eval '%_defaultdocdir')
+cat "${RPMTEST}${RPM_DOCDIR}/rpm/macros.rpmsign-sequoia" >> ${HOME}/.rpmmacros
+cat << EOF >> ${HOME}/.rpmmacros
+%_gpg_name 771B18D3D7BAA28734333C424344591E1964C5FC
+EOF
+
+runroot_other sq key import /data/keys/*.secret
+runroot rpmkeys --import /data/keys/*.pub
+],
+[0],
+[ignore],
+[ignore])
+
+RPMTEST_CHECK([
+runroot rpmbuild -bb --quiet \
+	--define "_rpmfilever 6" \
```

Comment:
This obviously wont produce v6 packages in 4.x, we'll need to ship a pre-built v6 package for testing purposes.